rpcserver.go

```diff
@@ -6748,6 +6754,76 @@ func (r *rpcServer) ListPermissions(_ context.Context,
 	}, nil
 }
 
+// ListPermissions lists all RPC method URIs and their required macaroon
+// permissions to access them.
+func (r *rpcServer) UpdateNodeAnnouncement(_ context.Context,
+	req *lnrpc.NodeAnnouncementUpdateRequest) (
+	*lnrpc.NodeAnnouncementUpdateResponse, error) {
+
+	nodeModifiers := make([]netann.NodeAnnModifier, 0)
+
+	// TODO(positiveblue): add Features modifier
+
+	if req.Color != "" {
+		color, err := parseHexColor(req.Color)
+		if err != nil {
+			return nil, fmt.Errorf("unable to parse color: %v", err)
+		}
+
+		nodeModifiers = append(nodeModifiers, netann.NodeAnnSetColor(color))
+	}
+
+	if req.Alias != "" {
+		alias, err := lnwire.NewNodeAlias(req.Alias)
+		if err != nil {
+			return nil, fmt.Errorf("invalid alias value: %v", err)
+		}
+		nodeModifiers = append(nodeModifiers, netann.NodeAnnSetAlias(alias))
+	}
+
+	if len(req.Addresses) > 0 {
+		// TODO(positiveblue): add address validation
+		var newAddrs []net.Addr
+
+		for _, addr := range req.Addresses {
+			newAddr, err := net.ResolveTCPAddr(addr.Network, addr.Addr)
+			if err != nil {
+				return nil, fmt.Errorf("unable to resolve "+
+					"address %v: %v", addr, err)
+			}
+			newAddrs = append(newAddrs, newAddr)
+		}
+
+		nodeModifiers = append(nodeModifiers, netann.NodeAnnSetAddrs(newAddrs))
+	}
+
+	if len(nodeModifiers) == 0 {
+		return nil, fmt.Errorf("unable detect any new values to update " +
+			"the node announcement")
+	}
+
+	// Then, we'll generate a new timestamped node
+	// announcement with the updated addresses and broadcast
+	// it to our peers.
+	newNodeAnn, err := r.server.genNodeAnnouncement(
```

Comment:
So we'll end up signing a new valid node announcement, but this doesn't include writing the new node announcement to disk. If we don't write it to disk, then certain other sub-systems will become out of sync, as we'll have one we advertise in memory, and one that's read from disk.

Reply:
Good catch, I guess that I should perform then a `localChanDB.ChannelGraph().SetSourceNode(selfNode)` like the one [here](https://github.com/lightningnetwork/lnd/blob/master/server.go#L2470)